crates/turbo-tasks-memory/tests/task_input.rs

```diff
@@ -0,0 +1,182 @@
+use anyhow::Result;
+use turbo_tasks::{CompletionVc, TaskInput};
+use turbo_tasks_testing::{register, run};
+
+register!();
+
+#[derive(Clone, TaskInput, Eq, PartialEq, Debug)]
+struct NoFields;
+
+#[turbo_tasks::function]
+async fn no_fields(input: NoFields) -> CompletionVc {
+    // Can't fail.
+    assert_eq!(input, NoFields);
+    CompletionVc::immutable()
+}
+
+#[derive(Clone, TaskInput)]
+struct OneUnnamedField(u32);
+
+#[turbo_tasks::function]
+async fn one_unnamed_field(input: OneUnnamedField) -> CompletionVc {
+    assert_eq!(input.0, 42);
+    CompletionVc::immutable()
+}
```

Comment:
Nit: I think this test file would be much more straight forward if we didn't depend on turbo at all. Something like:

```suggestion
#[derive(TaskInput)]
struct OneUnnamedField(u32);

#[test]
fn test_one_unnamed_field() -> Result<()> {
    let input = OneUnnamedField(42);
    let actual: OneUnnamedField = FromTaskInput::try_from(&TaskInput::from(input))?;
    assert_eq!(actual.0, 42);
    Ok(())
}
```

Reply:
Fair enough. I'll move these tests to turbo-tasks. I'll just keep one here as an integration test between tt::function and derive(TaskInput).